### PR TITLE
EPMDJ-9407: Refactor Portal component

### PR DESCRIPTION
**Pull request by dmitrii-kliushentsev**

### CODE REVIEW

File: src/components/portal/portal.tsx [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
+    if (rootElementById) {
+      rootElementById.appendChild(element);
     }
-  }
+    return () => {
+      if (rootElementById) {
+        rootElementById.removeChild(element);
+      }
+    };
+  }, [element]);
```
Comment (IliaSolovev) [inline src/components/portal/portal.tsx:22]:
```suggestion
  }, [rootElementById]);
```

File: src/components/portal/portal.tsx [hunk 2]
```diff
@@ -1,31 +1,25 @@
...
+  displayContent: boolean,
+  children: ReactChild,
+}
 
-  public componentWillUnmount() {
-    if (this.rootElementById) {
-      this.rootElementById.removeChild(this.element);
+export const Portal = ({ rootElementId = '', displayContent, children }: Props): ReactPortal | null => {
+  const element = document.createElement('div');
+  const rootElementById = document.getElementById(rootElementId);
```
Comment (RomanAchilov) [inline src/components/portal/portal.tsx:12]: in a good way, the creation of the handler should take place inside useEffect

Comment (RomanAchilov) [reply src/components/portal/portal.tsx:12]: https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily
check this

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/components/portal/portal.tsx [hunk 3]
```diff
@@ -1,31 +1,29 @@
...
 
-  public componentWillUnmount() {
-    if (this.rootElementById) {
-      this.rootElementById.removeChild(this.element);
+export const Portal = ({ rootElementId = '', displayContent, children }: Props): ReactPortal | null => {
+  const element = useMemo(() => document.createElement('div'), []);
+  useEffect(() => {
+    const rootElementById = document.getElementById(rootElementId) || createAndAppendElementWithId(rootElementId);
+    if (rootElementById) {
+      rootElementById.appendChild(element);
```
Comment (IliaSolovev) [inline src/components/portal/portal.tsx:17]:
```suggestion
      rootElementById?.appendChild(element);
```

File: src/components/portal/portal.tsx [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
+    return () => element.remove();
+  }, [rootElementId]);
+  return displayContent ? createPortal(children, element) : null;
+};
 
-  public render() {
-    const { children, displayContent } = this.props;
-    return displayContent ? createPortal(children, this.element) : null;
-  }
+function createAndAppendElementWithId(id: string): HTMLDivElement {
```
Comment (IliaSolovev) [inline src/components/portal/portal.tsx:24]:
```suggestion
function createAndAppendElement(id: string): HTMLDivElement {
```

**Review by IliaSolovev — CHANGES_REQUESTED** [review]

File: src/components/portal/portal.tsx [hunk 5]
```diff
@@ -1,31 +1,22 @@
-import { Component } from 'react';
+import {
+  ReactChild, ReactPortal, useEffect, useMemo,
```
Comment (IliaSolovev) [inline src/components/portal/portal.tsx:2]: remove unused vars

**Review by IliaSolovev — APPROVED** [review]

**Pull request merged**